### ISSUE TITLE
[FEATURE] DataTable from_dataframe() 

### ISSUE THREAD

**Issue by thomktz** [bokeh/bokeh#12243]
### Problem description

Lack of easy way to create simple tables from pandas DataFrames, with some basic colouring or formatting options

### Feature description

Add a from_dataframe() method to the DataTable class

### Potential alternatives

A create_datatable_from_dataframe() standalone function

### Additional information

II can work on this myself if others are interested in the feature

**Comment by thomktz** [comment]
Similarly, a from_dict() method could be implemented.
Both should be static methods in order to work like this:

`table = DataTable.from_dataframe(df, formatters={'col1': my_formatter})`

**Comment by thomktz** [comment]
I made some tests on a fork

`    @staticmethod
    def from_dataframe(dataframe, formatters={}, editors={}, **kwargs) -> DataTable:
        """Creates a DataTable from the columns and values of a pandas DataFrame.
        
        formatters: Dictionnary of instantiated formatters. Keys must be in the DataFrame's columns.
        editors: Dictionnary of instantiated editors. Keys must be in the DataFrame's columns.

        """
        columns = [TableColumn(field=col, 
                               title=col, 
                               formatter=formatters[col] if col in formatters else CellFormatter(),
                               editor=editors[col] if col in editors else CellEditor()) for col in dataframe.columns]
        data_table = DataTable(columns=columns, source=ColumnDataSource(dataframe), **kwargs)
        return data_table`

This worked on a few exemples I tried, any insights ?

**Comment by bryevdv** [comment]
cc @bokeh/dev for any thoughts

**Comment by gmerritt123** [comment]
I think it's a good idea to have something like this. There is high level api to plot data on figures without explicitly knowing anything about ColumnDataSources e.g. p.line(x=[1,2,3],y=[1,2,3]) , why not something similar for a Table?

My only input/thoughts --> Ideally I'd like to be able to instantiate a Table model 2 different ways:

 1. The current way --> via a collection of TableColumns and pointing to a CDS
 2. Convenience/"high level" method to build one straight from from either a dataframe, a dictionary or a pre-existing CDS. With this method an optional arg specifying a subset of columns to use would be nice to have as well.

**Comment by thomktz** [comment]
I agree, manually creating columns is almost always useless when using DataFrames as a data source since the columns already have names. They can still be accessed by `table.columns` afterwards, and so are formatters with `table.columns[i].formatter`.
The high-level method should be able to take in either a formatter to be applied to all columns or a list/dictionnary of formatters

**Comment by bryevdv** [comment]
@Azaya89 is this something you might be interested to look into?

**Comment by Azaya89** [comment]
> @Azaya89 is this something you might be interested to look into?

Sorry for the late reply. This seems similar to the discussion we had a while ago. I could take a look at it in the next few days if that's OK?

**Comment by bryevdv** [comment]
@Azaya89 any time that works for you!
